src/vm-vcpu/src/vcpu/mod.rs

```diff
@@ -244,81 +245,91 @@ impl KvmVcpu {
+                                eprintln!("Failed to read from mmio");
                             }
-                        } else if addr == 0x060 || addr == 0x061 || addr == 0x064 {
-                            // Write at the i8042 port.
-                            // See https://wiki.osdev.org/%228042%22_PS/2_Controller#PS.2F2_Controller_IO_Ports
-                        } else if 0x070 <= addr && addr <= 0x07f {
-                            // Write at the RTC port.
-                        } else {
-                            // Write at some other port.
                         }
-                    }
-                    VcpuExit::IoIn(addr, data) => {
-                        if 0x3f8 <= addr && addr < (0x3f8 + 8) {
-                            // Read from the serial port.
+                        VcpuExit::MmioWrite(addr, data) => {
                             if self
                                 .device_mgr
                                 .lock()
                                 .unwrap()
-                                .pio_read(PioAddress(addr), data)
+                                .mmio_write(MmioAddress(addr), data)
                                 .is_err()
                             {
-                                eprintln!("Failed to read from serial port");
+                                eprintln!("Failed to write to mmio");
                             }
-                        } else {
-                            // Read from some other port.
                         }
-                    }
-                    VcpuExit::MmioRead(addr, data) => {
-                        if self
-                            .device_mgr
-                            .lock()
-                            .unwrap()
-                            .mmio_read(MmioAddress(addr), data)
-                            .is_err()
-                        {
-                            eprintln!("Failed to read from mmio");
+                        _ => {
+                            // Unhandled KVM exit.
                         }
                     }
-                    VcpuExit::MmioWrite(addr, data) => {
-                        if self
-                            .device_mgr
-                            .lock()
-                            .unwrap()
-                            .mmio_write(MmioAddress(addr), data)
-                            .is_err()
-                        {
-                            eprintln!("Failed to write to mmio");
-                        }
-                    }
-                    _ => {
-                        // Unhandled KVM exit.
+                }
+                Err(e) => {
+                    // During boot KVM can exit with `EAGAIN`. In that case, do not
+                    // terminate the run loop.
+                    if e == Errno::new(libc::EAGAIN) {
+                    } else {
+                        eprintln!("Emulation error: {}", e);
+                        break;
```

Comment:
nit
```suggestion
                    if e != Errno::new(libc::EAGAIN) {
                        eprintln!("Emulation error: {}", e);
                        break;
                    }
```

Reply:
In the next commits this is updated to account for libc::EINTR as well. If that's fine with you, I'll not update it in this commit as well.